### ISSUE TITLE
Players cannot connect, error message "Disconnected"

### ISSUE THREAD

**Issue by Namnodorel** [SpigotMC/BungeeCord#1983]
We have a Spigot server, which should be running 24/7. However, after some time (maybe a day of uptime, it's hard to say) the server is shown as offline to the players and if one tries to join nothing but the error message "Disconnected" appears (with the server itself still running). This is really annoying, because we have to restart the server each time this happens manually. The exceptions on both server- and clientside are very uninformative:

When the client tries to ping the server:

```
Can't ping our.server.ip:25565: Internal Exception: io.netty.handler.codec.DecoderException:
com.google.gson.JsonSyntaxException: 
com.google.gson.stream.MalformedJsonException: 
Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 1
```

When the client tries to actually connect, "Logging in" is shown for an unusual long time, and except for the "Disconnected", no exception is thrown, the console output is only

> Connecting to our.server.ip, 25565

The serverside exception:
https://codeshare.io/63Jjw

Any help appreciated!

**Comment by kamcio96** [comment]
Did you update jar in a runtime?

**Comment by Namnodorel** [comment]
We have a hosted server, so maybe our hoster uses a script to update everything (to which we don't have access). We don't know really much about it, so it could have updated something while the server was running. But why is that important?

**Comment by kamcio96** [comment]
http://stackoverflow.com/a/14339210 :wink:

**Comment by Namnodorel** [comment]
Well, even if that caused an issue, I think it wouldn't persist after a restart... But we encounter it again and again and again...

**Comment by electronicboy** [comment]
Wherever you took that serverside log from, you need to replace the jar file as it appears to have issues. Either that or, which with the other error message you posted, you have a bad plugin on the server.

**Comment by Namnodorel** [comment]
We don't think it's a bad plugin, otherwise the exception would hint which one. And replacing the jar can be hard on external hosted servers because we'd need to contact the hoster to do that for us... And until it is not completely sure that the problem is a corrupted jar, I'd prefer to wait with this step.

**Comment by electronicboy** [comment]
Not always, plugins can cause issues when they bundle stuff with them, e.g. if you have a plugin that bundles netty, it can cause issues. This isn't an issue with Spigot/Bungee, this is likely an issue with your setup.

For the ping, that's likely a motd plugin throwing out an invalid message, or the joys of the classes on your server getting messed up, as mentioned in the next section.

On the client, it's getting stuck because the client starts to connect an the server stops sending anything. On the server, it fails to find the class, which the only time that can really only happen is from a bad jar or something messing up the server (e.g. like potentially speculated above, the jar file getting replaced while it's running. if they are using a script that automatically updates your jar files, that could cause the type of breakage you're seeing (and if they are, I'd consider swapping hosts, such a setup would indicate they have no idea what they're doing and will likely break your server from doing stuff like that)).

Basically, this ticket is mostly pointless right now, you either have a bad plugin, or you need to speak to your hosting provider.

**Comment by i9hdkill** [comment]
^
